AlfrescoDocumentPickerFileProvider/FileProvider.m

```diff
@@ -464,34 +463,36 @@ - (void)stopProvidingItemAtURL:(NSURL *)url
 
 - (void)importDocumentAtURL:(NSURL *)fileURL toParentItemIdentifier:(NSFileProviderItemIdentifier)parentItemIdentifier completionHandler:(void (^)(NSFileProviderItem _Nullable, NSError * _Nullable))completionHandler
 {
-    AlfrescoFileProviderItemIdentifierType typeIdentifier = [AFPItemIdentifier itemIdentifierTypeForIdentifier:parentItemIdentifier];
-    if(typeIdentifier == AlfrescoFileProviderItemIdentifierTypeLocalFiles)
+    [fileURL startAccessingSecurityScopedResource];
+    NSError *error = nil;
+    NSDictionary *resourceValues = [fileURL resourceValuesForKeys:@[NSURLNameKey, NSURLCreationDateKey, NSURLContentModificationDateKey, NSURLTypeIdentifierKey, NSURLTotalFileSizeKey] error:&error];
+    if(error)
     {
-        [fileURL startAccessingSecurityScopedResource];
-        NSError *error = nil;
-        NSDictionary *resourceValues = [fileURL resourceValuesForKeys:@[NSURLNameKey, NSURLCreationDateKey, NSURLContentModificationDateKey, NSURLTypeIdentifierKey, NSURLTotalFileSizeKey] error:&error];
-        if(error)
+        completionHandler(nil, error);
+    }
+    else
+    {
+        AFPItem *item = [[AFPItem alloc] initWithImportedDocumentAtURL:fileURL resourceValues:resourceValues parentItemIdentifier:parentItemIdentifier];
+        
+        AlfrescoFileProviderItemIdentifierType typeIdentifier = [AFPItemIdentifier itemIdentifierTypeForIdentifier:parentItemIdentifier];
+        if(typeIdentifier == AlfrescoFileProviderItemIdentifierTypeLocalFiles)
         {
-            completionHandler(nil, error);
+            [self.fileService saveToLocalFilesDocumentAtURL:fileURL];
+            [fileURL stopAccessingSecurityScopedResource];
         }
-        else
+        else if(typeIdentifier == AlfrescoFileProviderItemIdentifierTypeSyncFolder)
         {
-            NSString *downloadContentPath = [[AlfrescoFileManager sharedManager] downloadsContentFolderPath];
-            NSString *fullDestinationPath = [downloadContentPath stringByAppendingPathComponent:resourceValues[NSURLNameKey]];
-            NSURL *destinationURL = [NSURL fileURLWithPath:fullDestinationPath];
-            
-            AFPItem *item = [[AFPItem alloc] initWithImportedDocumentAtURL:fileURL resourceValues:resourceValues parentItemIdentifier:parentItemIdentifier];
-            
-            [self saveDocumentAtURL:fileURL toURL:destinationURL overwritingExistingFile:NO];
-            [fileURL stopAccessingSecurityScopedResource];
-            [[NSFileProviderManager defaultManager] signalEnumeratorForContainerItemIdentifier:parentItemIdentifier completionHandler:^(NSError * _Nullable error) {
-                completionHandler(item, error);
-            }];
+            NSURL *storageURL = [item fileURL];
+            NSFileManager *fileManager = [NSFileManager new];
```

Comment:
You could use the defaultManager convenience method to avoid allocating new memory